Source/FakeItEasy/Creation/FakeOptions.cs

```diff
@@ -29,6 +30,17 @@ public IEnumerable<CustomAttributeBuilder> AdditionalAttributes
 
         public void AddInterfaceToImplement(Type interfaceType)
         {
+            Guard.AgainstNull(interfaceType, "interfaceType");
+
+            if (!interfaceType.IsInterface)
+            {
+                throw new ArgumentException(
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        "The specified type '{0}' is not an interface",
+                        interfaceType.Name));
+            }

```

Comment:
I feel terrible for not having thought of this before, but we can probably do better than just `Name`.
Here's a sample failure message when we just use `Name` and try to use `List<Foo>` as an interface (which it's not):

```
System.ArgumentException : The specified type 'List`1' is not an interface
```

However, there's an extension method, `FullNameCSharp`, which will give this output:

```
System.ArgumentException : The specified type 'System.Collections.Generic.List<FakeItEasy.Tests.Foo>' is not an interface
```

How do you feel about that?